### PR TITLE
[vcpkg baseline][liblzf] Move header lzf.h to include/liblzf

### PR DESCRIPTION
**Pull request by WangWeiLin-MV**
### Fix
Install port `liblzf` failed with conflict `include/lzf.h` installed by `h5py-lzf`.

### Change
Update `liblzf` installation header `lzf.h` from `include` to `include/liblzf`.

### Checklist
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [x] SHA512s are updated for each updated download.
- [ ] ~The "supports" clause reflects platforms that may be fixed by this new version.~
- [ ] ~Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.~
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### Test
Port usage tests pass with following triplets:
* x64-windows

### CODE REVIEW

File: ports/liblzf/CMakeLists.txt [hunk 1]
```diff
@@ -21,7 +21,7 @@ install(TARGETS liblzf
     LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
     ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
 )
-install(FILES lzf.h DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
+install(FILES lzf.h DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/liblzf)
```
Comment (dg0yt) [inline ports/liblzf/CMakeLists.txt:24]: Without changes to the export config, users will have to change
~~~
-#include <lzf.h>
+#include <liblzf/lzf.h>
~~~

**Comment by dg0yt** [comment]
Maybe it is better to change h5py

**Comment by WangWeiLin-MV** [comment]
> Maybe it is better to change h5py-lzf. AFAIU
> 
> * lzf there is an outdate vendored copy.
> * lzf there is not the primary (public) interface.
> 
> So if lzf cannot be devendored (to the new port) there, maybe the header can be moved to a subdir there.

Thanks for your suggestion, this is more appropriate. I will change the installation path of `h5py-lzf` to `include/h5py`

**Comment by dg0yt** [comment]
Originally I suggested to only move the conflicting header `lzf.h`, not the main header `lzf_filter.h`. But I start to realize that h5py-lzf is a pure vcpkg artifact. Upstream's aim is to provide an extension to HDF5, not liblzf. AFAIU `register_lzf(void)` is the only public interface of h5py-lzf. So the port probably shouldn't install `lzf.h` at all. Users who want the liblzf API must use the new liblzf port.

File: ports/liblzf/CMakeLists.txt (hunk 1 above)
Comment (WangWeiLin-MV) [reply ports/liblzf/CMakeLists.txt:24]: Updated.

**Comment by WangWeiLin-MV** [comment]
Update `liblzf` installation headers to `include/{PORT}`, and it was a workaround to avoid conflict.

Update `h5py-lzf` installation headers block with local test failed, I will fix this later.

**Comment by dg0yt** [comment]
See #37332.

**Comment by WangWeiLin-MV** [comment]
> Update `liblzf` installation headers to `include/{PORT}`, and it was a workaround to avoid conflict.
> 
> Update `h5py-lzf` installation headers block with local test failed, I will fix this later.

Thanks @dg0yt  for #37332, the incorrect dependencies of `h5py-lzf` has been fixed and local test pass.

So, closing this in favor of #37332

**Pull request closed without merging**